front_end_pocket_poll/lib/widgets/category_popup_single.dart

```diff
@@ -55,94 +51,79 @@ class _CategoryPopupSingleState extends State<CategoryPopupSingle> {
-                        // this shouldn't ever be reached, but put it here to be safe
-                        return Text("Error: ${snapshot.error}");
-                      } else {
-                        return Center(child: CircularProgressIndicator());
-                      }
-                    }),
-              ),
+                  child: (this.categoryRows.isNotEmpty)
+                      ? Container(
+                          height: MediaQuery.of(context).size.height * .25,
+                          key: Key("category_popup_single:category_container"),
+                          child: ListView.builder(
+                            shrinkWrap: true,
+                            itemCount: this.categoryRows.length,
+                            itemBuilder: (context, index) {
+                              return this.categoryRows[index];
+                            },
+                          ),
+                        )
+                      : Container(
+                          height: MediaQuery.of(context).size.height * .25,
+                          child: RichText(
+                            text: TextSpan(
+                              style: TextStyle(
+                                  color: (Globals.user.appSettings.darkTheme)
+                                      ? Colors.white
+                                      : Colors.black),
+                              children: [
+                                TextSpan(
+                                    text:
+                                        "No categories found in this group. Click on this icon:  "),
+                                WidgetSpan(
+                                  child: Icon(Icons.settings),
+                                ),
+                                TextSpan(
+                                    text:
+                                        " found in the top right corner of the group's page to add some."),
+                              ],
+                            ),
+                          ))),
             ],
           ),
         ),
       ),
     );
   }
 
+  // builds choice rows using the current group group categories
+  void buildChoiceRows() {
+    int index = 0; // used for integration testing
+    for (String catId in Globals.currentGroup.categories.keys) {
+      this.categoryRows.add(CategoryRow(
+            new Category(
+                categoryName:
+                    Globals.currentGroup.categories[catId].categoryName,
+                owner: Globals.currentGroup.categories[catId].owner),
+            this.selectedCategory != null &&
+                this.selectedCategory.categoryId == catId,
+            onSelect: () => selectCategory(new Category(
+                categoryId: catId,
+                categoryName:
+                    Globals.currentGroup.categories[catId].categoryName)),
+            index: index,
+          ));
+      index++;
+    }
```

Comment:
This is just a preference thing, do what you want.

I think having to enter 'Globals.currentGroup.categories[catId]' gets cumbersome in this block of code and it makes things hard to read. To fix this, you could loop over the [entries](https://api.dart.dev/stable/2.8.2/dart-core/Map/entries.html) of the Gloabls.currentGroup.categories map which would simplify the code a lot.

i.e.
```
    for (MapEntry<String, GroupCategory> catEntry
        in Globals.currentGroup.categories.entries) {
      this.categoryRows.add(CategoryRow(
            new Category(
                categoryName: catEntry.value.categoryName,
                owner: catEntry.value.owner),
            this.selectedCategory != null &&
                this.selectedCategory.categoryId == catEntry.key,
            onSelect: () => selectCategory(new Category(
                categoryId: catEntry.key,
                categoryName: catEntry.value.categoryName)),
            index: index,
          ));
      index++;
    }
```